pyEDAA/UCIS/Cobertura.py

```diff
@@ -49,43 +49,52 @@
 class Class:
 	"""Represents a code element in the Cobertura coverage data model (Java-focused)."""
 
-	source_file: str
+	name: str
+	sourceFile: str
 	lines: Dict[int, int]
-	lines_valid: int
-	lines_covered: int
+	linesValid: int
+	linesCovered: int
 
-	def __init__(self, source_file: str):
-		self.source_file = source_file
+	def __init__(self, name: str, sourceFile: str):
+		self.name = name
+		self.sourceFile = sourceFile
 		self.lines = {}
-		self.lines_valid = 0
-		self.lines_covered = 0
+		self.linesValid = 0
+		self.linesCovered = 0
 
-	def add_statement(self, line: int, count: int) -> None:
-		assert line not in self.lines.keys()
-		self.lines[line] = count
-		self.lines_valid += 1
-		if count:
-			self.lines_covered += 1
-
-	def get_xml_node(self) -> etree.Element:
-		class_node = etree.Element("class")
-		class_node.attrib["name"] = self.source_file
-		class_node.attrib["filename"] = self.source_file
-		class_node.attrib["complexity"] = "0"
-		class_node.attrib["branch-rate"] = "0"
-		class_node.attrib["line-rate"] = str(self.lines_covered / self.lines_valid)
-		class_node.append(etree.Element("methods"))
-		lines_node = etree.SubElement(class_node, "lines")
+	def addStatement(self, line: int, hits: int) -> None:
+		assert line not in self.lines.keys(), "Duplicated line number"
+
+		self.lines[line] = hits
+		self.linesValid += 1
+
+		if hits:
+			self.linesCovered += 1
+
+	def getXmlNode(self) -> etree.Element:
+		classNode = etree.Element("class")
+		classNode.attrib["name"] = self.sourceFile
+		classNode.attrib["filename"] = self.sourceFile
+		classNode.attrib["complexity"] = "0"
+		classNode.attrib["branch-rate"] = "0"
+
+		try:
+			classNode.attrib["line-rate"] = str(self.linesCovered / self.linesValid)
```

Comment:
This should be refactored to
1. division and error handling
2. assignment and string conversion

```py
try:
  rate = self.linesCovered / self.linesValid
except ZeroDivisionError:
  rate = 1

classNode.attrib["line-rate"] = str(rate)
```
This reduces code duplication. Moreover, the `try..except` block surrounds only a minimal code portion, where it's needed.

Reply:
Fixed in https://github.com/edaa-org/pyEDAA.UCIS/pull/7/commits/a3989380939cc40dad5c58f093cfe2d9e3aa499f

Reply:
Checked.